ProcessLib/PhaseField/PhaseFieldFEM.h

```diff
@@ -0,0 +1,671 @@
+
+    PhaseFieldLocalAssembler(
+        MeshLib::Element const& e,
+        std::size_t const /*local_matrix_size*/,
+        bool const is_axially_symmetric,
+        unsigned const integration_order,
+        PhaseFieldProcessData<DisplacementDim>& process_data)
+        : _process_data(process_data),
+          _integration_method(integration_order),
+          _element(e),
+         _is_axially_symmetric(is_axially_symmetric)
+    {
+        unsigned const n_integration_points =
+            _integration_method.getNumberOfPoints();
+
+        _ip_data.reserve(n_integration_points);
+        _secondary_data.N.resize(n_integration_points);
+
+        auto const shape_matrices =
+            initShapeMatrices<ShapeFunction, ShapeMatricesType,
+                              IntegrationMethod, DisplacementDim>(
+                e, is_axially_symmetric, _integration_method);
+
+        SpatialPosition x_position;
+        x_position.setElementID(_element.getID());
+
+        for (unsigned ip = 0; ip < n_integration_points; ip++)
+        {
+            // displacement (subscript u)
+            _ip_data.emplace_back(*_process_data.material);
+            auto& ip_data = _ip_data[ip];
+            ip_data.integration_weight =
+                _integration_method.getWeightedPoint(ip).getWeight() *
+                shape_matrices[ip].integralMeasure * shape_matrices[ip].detJ;
+
+            ip_data.sigma.setZero(kelvin_vector_size);
+            ip_data.sigma_prev.resize(kelvin_vector_size);
+            ip_data.eps.setZero(kelvin_vector_size);
+            ip_data.eps_prev.resize(kelvin_vector_size);
+            ip_data.C_tensile.setZero(kelvin_vector_size, kelvin_vector_size);
+            ip_data.C_compressive.setZero(kelvin_vector_size,
+                                         kelvin_vector_size);
+            ip_data.sigma_tensile.setZero(kelvin_vector_size);
+            ip_data.sigma_compressive.setZero(kelvin_vector_size);
+            // ip_data.strain_energy_tensile;
+            ip_data.history_variable =
+                process_data.history_field(0, x_position)[0];
+            ip_data.history_variable_prev =
+                process_data.history_field(0, x_position)[0];
+            ip_data.sigma_real.setZero(kelvin_vector_size);
+            // ip_data.sigma_real_prev.setZero(kelvin_vector_size);
+
+            ip_data.N = shape_matrices[ip].N;
+            ip_data.dNdx = shape_matrices[ip].dNdx;
+
+            _secondary_data.N[ip] = shape_matrices[ip].N;
+        }
+    }
+
+    void assemble(double const /*t*/, std::vector<double> const& /*local_x*/,
+                  std::vector<double>& /*local_M_data*/,
+                  std::vector<double>& /*local_K_data*/,
+                  std::vector<double>& /*local_rhs_data*/) override
+    {
+        OGS_FATAL(
+            "PhaseFieldLocalAssembler: assembly without jacobian is not "
+            "implemented.");
+    }
+
+    void assembleWithJacobian(double const t,
```

Comment:
Documentation is needed for this assembly. You can use the attached PDF. To do this
1. Set HTML_EXTRA_FILES in Doxyfile.in (https://github.com/xingyuanmiao/ogs/blob/phasefield_wip/Documentation/Doxyfile.in#L1137) as
    HTML_EXTRA_FILES =  ../../web/content/docs/benchmarks/phase-field/Miao_Biot2017.pdf
2. Cite the PDF with a link to the pdf, for example,  ``<a href="Miao_Biot2017.pdff" target="_blank"><b>Phase field method</b></a>``.